WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/usecases/FetchScanStateUseCase.kt

```diff
@@ -27,32 +26,36 @@ class FetchScanStateUseCase @Inject constructor(
     suspend fun fetchScanState(
         site: SiteModel,
         polling: Boolean = true,
-        delayInMs: Long = FETCH_SCAN_STATE_DELAY_MILLIS
+        delayInMs: Long = FETCH_SCAN_STATE_DELAY_MILLIS,
```

Comment:
Warning (⚠️): I understand that you added this `delayInMs` as a parameter in order to be able to override that within its test suite. Note that this is problematic as it is generally advised to not do that, meaning adding functionality to the production code for test to work. For example, in this case, this allows clients to trigger the `fetchScanState(...)` function with any delay they like, which might cause confusion for future developers, since we actually don't want to allow that.

There is a really nice test-only related solution to this coroutines related problem that I advised Annmarie to follow as well (see below). I actually did the same on my own PR for that matter.

Test-only solution: Consider using the `MainCoroutineScopeRule` approach to fake the delay, by replacing `test { ... }` with `coroutineScope.runBlockingTest { ... }` and then add the `advanceTimeBy(DELAY_MILLIS)` right after the test act line. There are a couple of tests already using this approach, please search for usages of `MainCoroutineScopeRule` to find out more about it.

Reply:
Nice suggestion 👍
In 1243a65, bd5d4a2

I've used `MainCoroutineScopeRule` earlier, this is indeed cleaner!